lib/Doctrine/Migrations/Tools/TransactionHelper.php

```diff
@@ -59,13 +60,30 @@ public static function rollbackIfInTransaction(Connection $connection): void
 
     private static function inTransaction(Connection $connection): bool
     {
+        $innermostConnection = self::getInnerConnection($connection);
+
+        /* Attempt to commit or rollback while no transaction is running
+           results in an exception since PHP 8 + pdo_mysql combination */
+        return ! $innermostConnection instanceof PDO || $innermostConnection->inTransaction();
+    }
+
+    /**
+     * @return Connection|\Doctrine\DBAL\Driver\Connection|object|resource|null
+     */
+    private static function getInnerConnection(Connection $connection)
+    {
+        if (method_exists($connection, 'getNativeConnection')) {
+            try {
+                return $connection->getNativeConnection();
+            } catch (LogicException $e) {
```

Comment:
should we handle this case and ignore it? It might happen if some wrapped connection class does not implement the new method yet...

Reply:
This is fine, I guess.